File: 백준/Silver/5397. 키로거/키로거.py
def find_password(test_cases):
    results = []

    for keys in test_cases:
        left_stack = []  # 커서 왼쪽의 문자들
        right_stack = []  # 커서 오른쪽의 문자들

        for key in keys:
            if key == '-':
                # 백스페이스: 왼쪽 스택의 마지막 문자를 제거
                if left_stack:
                    left_stack.pop()
            elif key == '<':
                # 커서 왼쪽 이동: 왼쪽 스택에서 오른쪽 스택으로 이동
                if left_stack:
                    right_stack.append(left_stack.pop())
            elif key == '>':
                # 커서 오른쪽 이동: 오른쪽 스택에서 왼쪽 스택으로 이동
                if right_stack:
                    left_stack.append(right_stack.pop())
            else:
                # 일반 문자 입력: 왼쪽 스택에 추가
                left_stack.append(key)

        # 최종 결과: 왼쪽 스택 + 오른쪽 스택 (오른쪽 스택은 뒤집어서 합침)
        results.append(''.join(left_stack) + ''.join(reversed(right_stack)))

    return results

if __name__ == "__main__":
    # 입력 받기
    t = int(input())  # 테스트 케이스 개수
    test_cases = [input().strip() for _ in range(t)]

    # 비밀번호 계산
    results = find_password(test_cases)

    # 출력
    for result in results:
        print(result)

File: 백준/Silver/5397. 키로거/키로거.rs
use std::io;

fn process_keylogger(input: &str) -> String {
    let mut left_stack: Vec<char> = Vec::new();
    let mut right_stack: Vec<char> = Vec::new();
    
    for ch in input.chars() {
         match ch {
             '-' =>  {
                 if !left_stack.is_empty() {
                     left_stack.pop();
                 }
             }
             '<' => {
                 if !left_stack.is_empty() {
                     right_stack.push(left_stack.pop().unwrap());
                 }
             }
             '>' => {
                 if !right_stack.is_empty() {
                     left_stack.push(right_stack.pop().unwrap());
                 }
             }
             _ => {
                 left_stack.push(ch);
             }
        }
    }
    
    left_stack.extend(right_stack.into_iter().rev());
    left_stack.into_iter().collect()
}

fn main() {
    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .unwrap();
    let t: usize = input.trim().parse().unwrap();
    
    let mut results = Vec::new();
    for _ in 0..t {
        let mut case = String::new();
        io::stdin()
            .read_line(&mut case)
            .unwrap();
        results
            .push(process_keylogger(case.trim()));
    }
    
    for result in results {
        println!("{}", result);
    }
}